query/physicalplan/physicalplan.go

```diff
@@ -14,10 +15,12 @@ import (
 	"github.com/polarsignals/frostdb/query/logicalplan"
 )
 
+var concurrency = runtime.NumCPU()
```

Comment:
I think this could be the default, but I think concurrency should be a query engine option

Reply:
In another branch I had exactly that TODO comment. Forgot to adding it here. :grimacing:

Reply:
Actually, this should be just `1` in this PR. We don't want concurrency yet, just multiple callbacks support.